pkg/controller/replicaset/replica_set.go

```diff
@@ -198,6 +219,19 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.ReplicaSet {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	obj, cached := rsc.lookupCache.GetMatchingObject(pod)
+	rs, ok := obj.(*extensions.ReplicaSet)
+	if !ok {
```

Comment:
OK - I think I understand it.
The problem is basically in TestDeleteFinalStateUnknown test. And what is happening is that this test:
- adds RS to the underlying store
- then tries to delete a pod, which wasn't previously added - so it's not in the cache.

So basically:
rsc.lookupCache.GetMatchingObject(pod) is returning nil

So the code in getPodReplicaSet should look like that:

obj, cached := rsc.lookupCache.GetMatchingObject(pod)
if cached {
  rs, ok := obj.(*extensions.ReplicaSet)
  if ok && rsc.isCacheValid(pod, rs) {
    return rs
  }
  if !ok {
    glog.Errorf("lookup cache does not retuen a ReplicaSet object")
  }
}

....